x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/date_format/index.tsx

```diff
@@ -16,9 +16,13 @@ import { SetupInitializer } from '../../../plugin';
 
 const { DateFormat: strings } = ArgumentStrings;
 
+const getDateFormatArgInput = (defaultDateFormats: ComponentProps['dateFormats']) => (
+  props: Omit<ComponentProps, 'dateFormats'> & Partial<Pick<ComponentProps, 'dateFormats'>>
```

Comment:
Am I right that using that construction you are trying to override types for `dateFormats`? 
Maybe we can use `Assign` instead? 

![image](https://user-images.githubusercontent.com/20072247/128181814-8da36134-953e-42a2-8cc2-c3df236ed5e6.png)

find examples by `import { Assign } from '@kbn/utility-types';`

Reply:
I'm trying to make `dateFormats` optional.

Reply:
`Assign<ComponentProps, {dateFormats?: ComponentProps['dateFormats']}>` ?

Reply:
Why not)

Reply:
Done.